### DISCUSSION TITLE
3D Model gets Label Node's Color

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#2048] [Support Q&A]
I use GL3 built of OSG and osgEarth master branch.
I load an .obj 3D Model, and I observe a strange behavior. 3D model gets the color of the closest text's color. You may see the behavior in the video. There is a blue colored Label Node below the aircraft, and another Label Node which is far away and white colored. When the blue Label Node is visible, aircraft gets the blue color. When blue Label Node is out of view, aircraft gets white color of the distant Label Node. Most probably shader of the text overrides color of the 3D Model. What can I do to prevent this?

https://user-images.githubusercontent.com/5169314/164565444-91bcd3a1-4353-46ca-9f3a-07e9321b839a.mp4

**Comment by 640kb** [comment]
I had similar effects once in another context (without osgEarth, but with osg). Do you have 3D objects in the scene that use transparency? If you have the option, I would take the transparency out and see if the effect still occurs.

If it's because of that, you have to adjust the order of rendering of some objects, as far as I know. You can adjust that via the osg::StateSet class in osg.

**Reply by nebsar** [reply]
Thank you I will look into that.
